comit/src/expiries/config.rs

```diff
@@ -0,0 +1,289 @@
+const BITCOIN_MINE_WITHIN_N_BLOCKS: usize = 3; // Value arbitrarily picked.
+const ETHEREUM_MINE_WITHIN_N_BLOCKS: usize = 3;
+
+// TODO: Add support for lightning.
+#[derive(Clone, Copy, Debug)]
+pub enum Protocol {
+    Herc20Hbit,
+    HbitHerc20,
+}
+
+impl Protocol {
+    pub fn config(&self) -> Config {
+        // Since we are targeting Nectar and the App we can set these.
+        let alice = ActorType::Human;
+        let bob = ActorType::Bot;
+
+        match self {
+            Protocol::Herc20Hbit => Config {
+                alice,
+                bob,
+                alpha_required_confirmations: ethereum_required_confirmations(),
+                beta_required_confirmations: bitcoin_required_confirmations(),
+                alpha_block_time: ETHEREUM_BLOCK_TIME_SECS,
+                beta_block_time: BITCOIN_BLOCK_TIME_SECS,
+                alpha_mine_fund_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                beta_mine_fund_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                alpha_mine_redeem_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                beta_mine_redeem_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+            },
+            Protocol::HbitHerc20 => Config {
+                alice,
+                bob,
+                alpha_required_confirmations: bitcoin_required_confirmations(),
+                beta_required_confirmations: ethereum_required_confirmations(),
+                alpha_block_time: BITCOIN_BLOCK_TIME_SECS,
+                beta_block_time: ETHEREUM_BLOCK_TIME_SECS,
+                alpha_mine_fund_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                beta_mine_fund_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+                alpha_mine_redeem_within_n_blocks: BITCOIN_MINE_WITHIN_N_BLOCKS,
+                beta_mine_redeem_within_n_blocks: ETHEREUM_MINE_WITHIN_N_BLOCKS,
+            },
+        }
+    }
+}
+
+/// Configuration values used during transition period calculations.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Config {
+    alice: ActorType,
+    bob: ActorType,
+    alpha_required_confirmations: usize,
+    beta_required_confirmations: usize,
+    alpha_block_time: usize,
+    beta_block_time: usize,
+    alpha_mine_fund_within_n_blocks: usize,
+    beta_mine_fund_within_n_blocks: usize,
+    alpha_mine_redeem_within_n_blocks: usize,
+    beta_mine_redeem_within_n_blocks: usize,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum ActorType {
+    Bot,
+    Human,
+}
+
+impl ActorType {
+    fn period_to_act(&self) -> Duration {
+        match self {
+            ActorType::Bot => 15.minutes(),
+            ActorType::Human => 60.minutes(),
+        }
+    }
+}
+
+impl Config {
+    // Optimisation: finality_alpha/beta never change, we should cache the value so
+    // we only do the calculation once.
+
+    // Open Question: Is time_to_mine_n_blocks() _really_ better than just doing:
```

Comment:
Yes because on Bitcoin the difficulty changes every 2016 blocks (2 weeks). Which means that if there is a big influx or withdraw of hashing power, the average block time could drastically change which would then influence this figure.

Reply:
We use average block time of 10 minutes so the code is still subject to the weakness you point out - hence the question.

Reply:
It's still more precise and if our hardcoding of 10min becomes an issue in the future we can optimise.

If you decide to move away from `time_to_mine_n_blocks` I think it's ok-ish too. We just need to take in account that 10min is around 30% approx. It'd be good to have a quick look at the historical because hashpower shift because has encountered (I think it was 30% at the last halving , hence my suggestion).